src/levanter/data/shard_cache.py

```diff
@@ -1744,10 +1767,30 @@ def with_batch_size(self, batch_size):
             self.cache_dir, batch_size, self._ledger, self._broker, self._reader_offset, self._num_readers
         )
 
+    def _scan_chunk(self, chunk):
+        num_batches = (chunk.num_rows + self._batch_size - 1) // self._batch_size
```

Comment:
so the chunk metadata knows exactly how many examples/rows are in each chunk

Also, I'm not sure that the batches from parquet are always the same size (e.g. for remainder at end). I'd be fine with remove the batch_size argument to this class

Reply:
I'm using the ChunkMetadata right now I believe! This skips the actual reading of the parquet file. Or are you saying there is a higher level metadata file I can read from to make this even faster?

Reply:
Empirically, this seek implementation drops my resume time for my big audio training jobs from hours to under a minute since the main bottleneck happens in the `reader = _ChunkReader.from_metadata(self.cache_dir, chunk, self._batch_size)` step.

Reply:
One thing that may be the source of confusion here -  the variable `chunk` is actually just the `ChunkMetadata` - I've updated the type signature to make that clearer.

I think it might be better to change the variable name entirely, but opted to be consistent with the original `_read_chunk` in which `chunk` also refers to a `ChunkMetadata` variable.

https://github.com/stanford-crfm/levanter/blob/main/src/levanter/data/shard_cache.py#L1754-L1757

Reply:
re: batches from parquet not always being the same size. I actually was a bit puzzled at this because you are right, it's not always divisible. 

Right now in the true data loading logic though, it looks as if the remainder at the end gets returns as it's own (smaller) batch, so I'm mirroring this in the seek logic:  https://github.com/stanford-crfm/levanter/blob/main/src/levanter/data/shard_cache.py#L1815-L1817